package/base-files/files/lib/upgrade/nand.sh

```diff
@@ -98,10 +97,10 @@ identify_tar() {
 
 nand_restore_config() {
 	sync
-	local ubidev=$( nand_find_ubi $CI_UBIPART )
-	local ubivol="$( nand_find_volume $ubidev rootfs_data )"
-	[ ! "$ubivol" ] &&
-		ubivol="$( nand_find_volume $ubidev $CI_ROOTPART )"
+	local ubidev=$(nand_find_ubi $CI_UBIPART)
+	local ubivol="$(nand_find_volume $ubidev rootfs_data)"
+	[ ! "$ubivol" ] \
```

Comment:
sounds like ! can be removed and && replaced with ||.

Reply:
wouldn't that require e.g. -n?

Reply:
No idea what you mean. It's just an if inversion.

```
[ "$ubivol" ] || ubivol="$(nand_find_volume $ubidev $CI_ROOTPART)"
```

Reply:
Don't we need `[ -n "$ubivol ]`?

Reply:
Hmm I think you're right. Alternatively, -z &&.

Reply:
:woman_dancing:

Reply:
the string by itself `[ "$ubivol" ]` is the same as -n `[ -n "$ubivol" ]`

Reply:
well, is it really the same? Have you checked all cases of "", undefined, whatever?

And if it really was, wouldn't -n still be much easier to understand and it's behavior be more well-known by most people?